### ISSUE TITLE
connecting multiple clients to a single server.

### ISSUE THREAD

**Issue by nwwells** [balderdashy/sails.io.js#62]
I'm trying to do some testing, and it appears that I can't connect to my server multiple times on the same port. Is this a known issue? Here's my test:

```
function setupConnection (client) {
  var socketIoClient = require('socket.io-client');
  var sailsIoClient = require('sails.io.js');

  var io = sailsIoClient(socketIoClient);
  // Set some options:
  // (you have to specify the host and port of the Sails backend when using 
  // this library from Node.js)
  io.sails.transports = ['websocket'];
  // disable verbose logging
  io.sails.environment = 'production';
  // disable auto-connect, so that we can connect to multiple domains
  io.sails.autoConnect = false;

  return io.sails.connect('http://'+client.tenantDomain+':1338');
}


describe('Multitenancy', function () {
  var client = company1;
  it('should be able to connect to ' + client.tenantDomain, function (done) {
    var socket = setupConnection(client);
    socket.get('/date', function (body) {
      console.log("/date:", body)
      socket.disconnect();
      done();
    });
  });

  it('should be able to connect to ' + client.tenantDomain, function (done) {
    var socket = setupConnection(client);
    socket.get('/date', function (body) {
      console.log("/date:", body)
      socket.disconnect();
      done();
    });
  });

  client = company2;
  it('should be able to connect to ' + client.tenantDomain, function (done) {
    var socket = setupConnection(client);
    socket.get('/date', function (body) {
      console.log("/date:", body)
      socket.disconnect();
      done();
    });
  });

  it('should be able to connect to ' + client.tenantDomain, function (done) {
    var socket = setupConnection(client);
    socket.get('/date', function (body) {
      console.log("/date:", body)
      socket.disconnect();
      done();
    });
  });
});
```

for brevity I left out the definition of `clients`, but I hope you can see what I'm trying to do. All my tests after the first timeout, and debugging shows that the sockets are in a disconnected state.

Also, FWIW, the docs [in the README.md](https://github.com/balderdashy/sails.io.js#disable-autoconnect-andor-connect-sockets-manually) show passing the URL as a part of the (single) options object, but it's actually the first parameter, and options are the second. tripped me up for a bit.

**Comment by sailsbot** [comment]
Thanks for posting, @nwwells.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by nwwells** [comment]
wait, seriously? because you guys haven't responded in 30 days you just close the issue, and I'm supposed to open a new one? This sucks.

**Comment by sgress454** [comment]
@nwwells I hear you.  The bot is something we're experimenting with, and it can be frustrating, but it's for a good cause.  Look at it this way: when you say:

> because you guys haven't responded in 30 days

who is "you guys", exactly?  Sails.js has a whole community around it, supposedly, but the signal-to-noise ratio of issues compared to the _tiny number of people who actually answer them_ makes it impossible to answer most posts.  The bot forces squeaky wheels to get greased.  And you have to admit, you squeaked.

Have you tried setting `io.sails.multiplex = false;`?  See https://github.com/socketio/socket.io-client#iourlstring-optsobjectsocket.
